### PR TITLE
CHE-718: Simplify workspace model & API

### PR DESCRIPTION
**Pull request by voievodin**
[Model changes](https://github.com/eclipse/che/commit/579db1c7260d09824e32ffbb4197e2f5fbadf72a): 
Removed - `UsersWorkspace`, `RuntimeWorkspace`
Added - `Workspace`, `WorkspaceRuntime`
Moved attributes from `WorkspaceConfig` to `Workspace`

[API & SPI changes](https://github.com/eclipse/che/commit/a6b326ec8c64a3cb2341bd2fb7a2763affc9d09c)
Simplified API by removing methods related to the runtime workspace instance.
Improved workspace sub-entity update methods.
Added `created` and `updated` attributes which allow client to track recent workspace activity.
Improved Workspace starting and stopping by synchronizing machines start queue.
Simplified `WorkspaceManager`, events moved to the `WorkspaceRuntimes`.
BadRequestException scope changed to service.

@gazarenkov, @garagatyi, @skabashnyuk, @akorneta  please review changes described above.

[Dashboard Adaptation](https://github.com/eclipse/che/commit/92ec59f81ffe4c0871a6bd1d79c172f823ba39e1)
@ashumilova, @benoitf please review the changes related to the dashboard.

### CODE REVIEW

**Comment by garagatyi** [comment]
LGTM

**Comment by garagatyi** [comment]
Code looks much better in many places. Nice job!

**Comment by akorneta** [comment]
ok

File: wsmaster/che-core-api-workspace/src/main/java/org/eclipse/che/api/workspace/server/WorkspaceService.java [hunk 1]
```diff
@@ -164,61 +179,48 @@ public Response create(@ApiParam(value = "The configuration to create the new wo
...
-    public UsersWorkspaceDto getByKey(@ApiParam(value = "Composite key", examples = @Example({@ExampleProperty("workspace12345678"),
-                                                                                              @ExampleProperty("username:workspace_name"),
-                                                                                              @ExampleProperty(":workspace_name")})) @PathParam("key") String key)
-            throws NotFoundException,
-                   ServerException,
-                   ForbiddenException,
-                   BadRequestException {
+    public WorkspaceDto getByKey(@ApiParam(value = "Composite key",
+                                           examples = @Example({@ExampleProperty("workspace12345678"),
+                                                                @ExampleProperty("username:workspace_name"),
```
Comment (sleshchenko) [inline wsmaster/che-core-api-workspace/src/main/java/org/eclipse/che/api/workspace/server/WorkspaceService.java]: It is not clear what is username for workspace. Maybe you should rewrite this with namespace?

**Comment by voievodin** [comment]
ci-build

**Comment by skabashnyuk** [comment]
ok

**Pull request merged**